tracee-ebpf/tracee/net_events.go

```diff
@@ -4,122 +4,169 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"github.com/aquasecurity/tracee/pkg/external"
 	"time"
 
 	"github.com/google/gopacket"
 	"inet.af/netaddr"
 )
 
+type PktMeta struct {
+	SrcIP    [16]byte `json:"src_ip"`
+	DestIP   [16]byte `json:"dest_ip"`
+	SrcPort  uint16   `json:"src_port"`
+	DestPort uint16   `json:"dest_port"`
+	Protocol uint8    `json:"protocol"`
+	_        [3]byte  //padding
+}
+
+func (t Tracee) parsePacketMetaData(payload *bytes.Buffer) (PktMeta, uint32, int, error) {
+	var pktMetaData PktMeta
+	var pktLen uint32
+
+	err := binary.Read(payload, binary.LittleEndian, &pktLen)
+	if err != nil {
+		return pktMetaData, 0, 0, err
+	}
+	var ifindex uint32
+	err = binary.Read(payload, binary.LittleEndian, &ifindex)
+	if err != nil {
+		return pktMetaData, 0, 0, err
+	}
+	interfaceIndex, ok := t.ngIfacesIndex[int(ifindex)]
+	if !ok {
+		return pktMetaData, 0, 0, err
+	}
+	err = binary.Read(payload, binary.LittleEndian, &pktMetaData)
+	if err != nil {
+		return pktMetaData, 0, 0, err
+	}
+	return pktMetaData, pktLen, interfaceIndex, nil
+}
+
+func isIpv6(ip [16]byte) bool {
+	zeroedPattern := make([]byte, 9, 9)
+	if bytes.Compare(ip[:9], zeroedPattern) == 0 {
+		return false
+	}
+	return true
+}
+
+//takes the packet metadata and create argument array with that data
+func createPacketMetadataArg(packetmeta PktMeta) []external.Argument {
```

Comment:
the function create the basic argument for the "net event " event . that is has now only one argument that is `external.PktMeta`.
the function takes a similar struct (`packetMeta`) and assigns some of it to the `extenal.PktMeta struct` 
the deafferents between the structs is firstly  the type of the Ip members, which has to be string in the extenal.PktMeta  (because of the print type we want to print ) struct and in the packetMeta has to byte array  (because of the binary. Read function) 

Secondly the  reason for the whole assigning in the `arg.SrcIp `and `arg.DstIp` is because im casing the ip from byte array to string and note that the functions `netaddr.IPFrom16` and  `netaddr.IPFrom4` expects for known size of byte  array which means i cant cast the whole byte array to size specific ( like to do something like that `netaddr.IPFrom4(packet.DestIP[:4]).String()' 

also the reason for having the `extenal.PktMeta` struct is for able Tracee rules to recognize that type

and BTW if you still thinks its great idea to move the ip checking and assign to another function, let me know (:

Reply:
If you want a general function for creating all the net events args, create one named `createNetEventsArgs` and call in it to a `createPacketMetadataArg`.
This way `createPacketMetadataArg` will return an `external.Argument` as expected, and the enveloping `createNetEventsArgs` will return `[]external.Argument` as expected of it.